playbot/bot.py

```diff
@@ -37,21 +38,36 @@ def on_welcome(self, c, e):
             time.sleep(0.5)
 
     def on_privmsg(self, c, e):
-        self.do_command(e, e.arguments[0], *e.arguments[1:])
+        e.target = re.sub("!.*","",e.source)
+        e.source = self.nickname
+        self.do_command(e)
 
     def on_pubmsg(self, c, e):
-        msg = e.arguments[0]
-        msg = msg.replace(':', ' ', 1)
-        msg = re.sub("[ \t]+", " ", msg).split(" ")
-        nick = msg[0]
-        if nick.lower() == self.connection.get_nickname().lower():
-            self.do_command(e, msg[1], *msg[2:])
-        return
+        # Double check nick
```

Comment:
In case we missed some bit of code that changed nick. I know its defensive coding but previously it was always using self.connection.get_nick. 

Basically we check the message starts with bot name then strip it off.